### PR TITLE
docs: add annotation resources

### PR DESCRIPTION
**Pull request by khurrammaqbool**
### This PR:

Related to https://github.com/Clinical-Genomics/project-planning/issues/158.

Added: annotation resources to the docs

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by khurrammaqbool** [comment]
@keyvanelhami, @ashwini06 
This PR is ready for review. Please let me know if I missed any annotation resource.

File: docs/balsamic_annotation.rst [hunk 1]
```diff
@@ -0,0 +1,254 @@
+***********************************
+BALSAMIC Annotation Resources
+***********************************
+
+BALSAMIC annotates somatic single nucleotide variants (SNVs) using ``ensembl-vep`` and ``vcfanno``. Somatic structural variants (SVs), somatic copy-number variants (CNVs) and germline single nucleotide variants are annotated using only ``ensembl-vep``. All SVs and CNVs are merged using ``SVDB`` before annotation for `Target Genome Analysis (TGA)` or `Whole Genome Sequencing (WGS)` analyses types.
```
Comment (ashwini06) [inline docs/balsamic_annotation.rst:5]:
```suggestion
BALSAMIC annotates somatic single nucleotide variants (SNVs) using ``ensembl-vep`` and ``vcfanno``. Somatic structural variants (SVs), somatic copy-number variants (CNVs) and germline single nucleotide variants are annotated using only ``ensembl-vep``. All SVs and CNVs are merged using ``SVDB`` before annotating for `Target Genome Analysis (TGA)` or `Whole Genome Sequencing (WGS)` analyses.
```

File: docs/balsamic_annotation.rst [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+`BALSAMIC` adds the following annotation from `ClinVar` database using ``vcfanno``.
+
+.. list-table:: ClinVar
+   :widths: 50 50
+   :header-rows: 1
+
+   * - VCF tag
+     - description
+   * - CLNACC
+     - calculated by self of overlapping values
```
Comment (keyvanelhami) [inline docs/balsamic_annotation.rst:29]: A bit unclear for me what this really means.

File: docs/balsamic_annotation.rst [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+   * - VCF tag
+     - description
+   * - COSMIC_CDS
+     - CDS annotation
+   * - COSMIC_GENE
+     - gene name
+   * - COSMIC_STRAND
+     - strand
+   * - COSMIC_CNT
+     - number of samples with this mutation
```
Comment (keyvanelhami) [inline docs/balsamic_annotation.rst:89]: nr of samples from where? The cosmic database?

File: docs/balsamic_annotation.rst [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+   * - COSMIC_AA
+     - peptide annotation
+
+
+Where relevant, `BALSAMIC` uses `ensembl-vep` to annotate somatic and germline SNVs and somatic SVs/CNVs from `1000genomes (phase3)`, `ClinVar`, `ESP, HGMD-PUBLIC`, `dbSNP`, `gencode`, `gnomAD`, `polyphen`, `refseq`, and `sift` databases.
```
Comment (keyvanelhami) [inline docs/balsamic_annotation.rst:94]: Please clarify "where relevant". When exactly will Balsamic apply this annotation?

File: docs/balsamic_annotation.rst [hunk 5]
```diff
@@ -0,0 +1,254 @@
+***********************************
+BALSAMIC Annotation Resources
```
Comment (keyvanelhami) [inline docs/balsamic_annotation.rst:2]: @ashwini06 do we have a link added in Balsamic doc for ``ensembl-vep`` and ``vcfanno``?

**Review by keyvanelhami — COMMENTED** [review]
Good work Khurram, I have just added some minor comments.

File: docs/balsamic_annotation.rst (hunk 5 above)
Comment (ashwini06) [reply docs/balsamic_annotation.rst:2]: @keyvanelhami : We have links for each balsamic tool in resources for eg: https://balsamic.readthedocs.io/en/latest/bioinfo_softwares.html#ensembl-vep https://balsamic.readthedocs.io/en/latest/bioinfo_softwares.html#vcfanno. Do you mean this links or something else?

File: docs/balsamic_annotation.rst (hunk 4 above)
Comment (khurrammaqbool) [reply docs/balsamic_annotation.rst:94]: It refers to both, a variant or variant type.

File: docs/balsamic_annotation.rst (hunk 3 above)
Comment (khurrammaqbool) [reply docs/balsamic_annotation.rst:89]: Yes.

File: docs/balsamic_annotation.rst (hunk 5 above)
Comment (khurrammaqbool) [reply docs/balsamic_annotation.rst:2]: I checked, the information is there.

File: docs/balsamic_annotation.rst (hunk 3 above)
Comment (ashwini06) [reply docs/balsamic_annotation.rst:89]: Maybe this description will make it more clear `number of samples having this somatic mutation in the COSMIC database`

Comment (keyvanelhami) [reply docs/balsamic_annotation.rst]: I agree with Ashwini

**Review by keyvanelhami — APPROVED** [review]

**Comment by khurrammaqbool** [comment]
Thank you @keyvanelhami and @ashwini06 for valuable suggestions.

**Pull request merged**